src/are-inputs-equal.ts

```diff
@@ -10,10 +10,14 @@ export default function areInputsEqual(
   // https://github.com/alexreardon/memoize-one/pull/59
 
   for (let i = 0; i < newInputs.length; i++) {
-    // using shallow equality check
-    if (newInputs[i] !== lastInputs[i]) {
+    if (hasChanged(newInputs[i], lastInputs[i])) {
       return false;
     }
   }
   return true;
 }
+
+// compare whether a value has changed, accounting for NaN
+function hasChanged(newValue: unknown, oldValue: unknown) {
+  return newValue !== oldValue && (newValue === newValue || oldValue === oldValue);
```

Comment:
Thanks so much! Do you think this would be clearer?

```ts
function hasChanged(newValue: unknown, oldValue: unknown): boolean {
  if(newValue === oldValue) {
    return false;
  }

  // Special case for NaN. NaN is not === to NaN
  if(Number.isNaN(newValue) && Number.isNaN(oldValue)) {
    return false;
  }
  
  return true;
}
```

It is a tiny bit more code though, but I think it is a bit clearer

Reply:
Yes, definitely. One problem that I see is that `Number.isNaN` isn't supported in ES5.

Reply:
Yes, definitely.